<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="model" default="Hydrus" /> 
  <arg name="type" default="hex" />
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg name="alpha" default="0.3" />
  <arg name="d_theta" default="0.01" />
  <arg name="grad_f_thresh" default="0.0001" />
  <arg name="extra_module_num" default="1" />
  <arg name="extra_module_link_num0" default="0" />
  <arg name="extra_module_link_num1" default="0" />
  <arg name="extra_module_mass0" default="0.5" />
  <arg name="extra_module_mass1" default="0.5" />
  <arg name="extra_module_offset0" default="0.3" />
  <arg name="extra_module_offset1" default="0.3" />
  <arg name="initial_theta0" default = "1.04" />
  <arg name="initial_theta1" default = "1.04" />
  <arg name="initial_theta2" default = "1.04" />
  <arg name="initial_theta3" default = "1.04" />
  <arg name="initial_theta4" default = "1.04" />
  <arg name="time_thresh" default = "10.0" />
  <arg name="v_thresh" default = "0.001" />
  <arg name="thread_num" default = "1" />
  <arg name="use_initial_theta" default="false" />
<node name="posture_optimization_node" pkg="hydrus_object_transportation" type="posture_optimization_node" respawn="false" launch-prefix="$(arg launch_prefix)" output="screen">
       <!-- for transform controller -->
    <rosparam file="$(find hydrus_transform_control)/config/hydrusx/$(arg type)/$(arg model).yaml" command="load" />
    <param name="alpha" value="$(arg alpha)" />
    <param name="d_theta" value="$(arg d_theta)" />
    <param name="grad_f_thresh" value="$(arg grad_f_thresh)" />
    <param name="use_initial_theta" value="$(arg use_initial_theta)" />
    <param name="initial_theta0" value="$(arg initial_theta0)" />
    <param name="initial_theta1" value="$(arg initial_theta1)" />
    <param name="initial_theta2" value="$(arg initial_theta2)" />
    <param name="initial_theta3" value="$(arg initial_theta3)" />
    <param name="initial_theta4" value="$(arg initial_theta4)" />
    <param name="extra_module_num" value="$(arg extra_module_num)" />
    <param name="extra_module_link_num0" value="$(arg extra_module_link_num0)" />
    <param name="extra_module_link_num1" value="$(arg extra_module_link_num1)" />
    <param name="extra_module_mass0" value="$(arg extra_module_mass0)" />
    <param name="extra_module_mass1" value="$(arg extra_module_mass1)" />
    <param name="extra_module_offset0" value="$(arg extra_module_offset0)" />
    <param name="extra_module_offset1" value="$(arg extra_module_offset1)" />
    <param name="time_thresh" value="$(arg time_thresh)" />
    <param name="v_thresh" value="$(arg v_thresh)" />
    <param name="thread_num" value="$(arg thread_num)" />
   </node>
</launch>
